### PR TITLE
Add training workflow to python package

### PR DESCRIPTION
**Pull request by carterbox**
Copied the Train class into a new training module and wrapped it with a small example function and CLI that loads some of Tao's pre-processed data. It seems to do a pretty good job on Tao's data.

Screenshot from 2022-12-12 15-17-44

I made minor changes to the implementations of the Train class, but a lot of changes to formatting and documentation.

I have also added a script to the test folder that can generate a simulated dataset. However, PtychoNN does not perform well on this dataset. I am trying to figure out if this is a scaling issue or if the simulated dataset is just harder because it uses natural images.

### CODE REVIEW

File: ptychonn/_train/__main__.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+
+    trainer.plotLearningRate(
+        save_fname=out_dir / 'learning_rate.svg',
+        show_fig=False,
+    )
+    ptychonn.plot.plot_metrics(
+        trainer.metrics,
+        save_fname=out_dir / 'metrics.svg',
+        show_fig=False,
+    )
```
Comment (carterbox) [inline ptychonn/_train/__main__.py]: @stevehenke, this is an example of using the Tester API.

**Review by stevehenke — APPROVED** [review]
This seems to be a nice training interface. The one part that probably needs some tweaking is the file I/O. In the cases where I'm not passing in the full file path, it would be helpful to return the full file path so the data can be transferred as part of a workflow. I'm sure that I'll have better feedback when I try to integrate with it. Good work!

File: ptychonn/_train/__main__.py (hunk 1 above)
Comment (stevehenke) [reply ptychonn/_train/__main__.py]: Thanks! This looks very reasonable.

**Comment by carterbox** [comment]
> . In the cases where I'm not passing in the full file path, it would be helpful to return the full file path so the data can be transferred as part of a workflow. 

I'm not exactly sure what you mean by "full file path", do you mean absolute path vs relative? In that case, you should convert your relative paths to absolute *before* passing them to this function. The standard library can do this for you.

It also occurred to me that you may want finer control over what is written to the disk, so I am going to allow passing `None` for all the filepath arguments. In that case, nothing is written to the disk in case you want to do a file-less workflow.

**Comment by carterbox** [comment]
- [x] Make file names accessible via parameter (programatically). Currently you have to read the docs and manually figure them out.

**Review by vbanakha — APPROVED** [review]
Looks good.

**Pull request merged**